### PR TITLE
Kbin sync 7 + lot of rework, refactoring / Mbin changes

### PR DESCRIPTION
**Pull request by melroy89**
- Finish Magazine deletion (Add TODO, since this can be done much easier and simpler when adding cascading to postgresql)
- Improve moderator role
  - Auto-formatting on YAML files according to your `.editorconfig`
- Improved cross-posted threads by grouping them
- Improve User profile visibility options  + account deletion process changed (but not yet how Mbin wants it)
- Keep our code most consistent eg. we use `VisibilityInterface::VISIBILITY_SOFT_DELETED;` instead of `self::`
- Rename Kbin class to Mbin class
- Add description to migration file (which is not present in Kbin)
- We move suspend/unsuspend feature to admin panel section
- Show to the user that the account is suspended
- Show avatar/cover (user box) above all the profile pages. Which make sense to me.
- Don't allow the user to post a thread, comment or new links, images or magazines, etc.
- Small code improvements / refactoring
- Fix people menu active class activation

Fixes: #194 

Kbin PRs:
- https://codeberg.org/Kbin/kbin-core/pulls/1207
- https://codeberg.org/Kbin/kbin-core/pulls/1209
- https://codeberg.org/Kbin/kbin-core/pulls/1210
- https://codeberg.org/Kbin/kbin-core/pulls/1211
- https://codeberg.org/Kbin/kbin-core/pulls/1213
- https://codeberg.org/Kbin/kbin-core/pulls/1214

### CODE REVIEW

File: templates/components/magazine_box.html.twig [hunk 1]
```diff
@@ -2,7 +2,11 @@
     {% if showSectionTitle %}
         <h3>{{ 'magazine'|trans }}</h3>
     {% endif %}
+<<<<<<< HEAD
```
Comment (e-five256) [inline templates/components/magazine_box.html.twig:5]: a leftover conflict here

Comment (melroy89) [reply templates/components/magazine_box.html.twig]: Fixed ;P

File: config/kbin_routes/admin.yaml [hunk 2]
```diff
@@ -1,37 +1,67 @@
...
+    controller: App\Controller\Admin\AdminModeratorController::moderators
+    path: /admin/moderators
+    methods: [GET, POST]
+
+admin_moderator_purge:
+    controller: App\Controller\Admin\AdminModeratorController::removeModerator
+    path: /admin/moderators/purge/{username}
+    methods: [POST]
+
+admin_deletion_users:
```
Comment (e-five256) [inline config/kbin_routes/admin.yaml:48]: ln 48-56 got duplicated, looks like they can be removed, they're declared at the top of the changes

Comment (melroy89) [reply config/kbin_routes/admin.yaml:48]: Interesting why git merged that wrong.

**Comment by e-five256** [comment]
I'm wondering if we should make these separate PRs. global moderators seems fine, not sure about the crossposting one, might need someone with federation to look at that, but RTR2 has a ton of issues, aside from just not working at all anyways (all your posts still show up if someone looks at your profile) and is abusable (can still post and generate notifications for people with no way for them to view / moderate / report the post)

**Comment by melroy89** [comment]
> I'm wondering if we should make these separate PRs. global moderators seems fine, not sure about the crossposting one, might need someone with federation to look at that, but RTR2 has a ton of issues, aside from just not working at all anyways (all your posts still show up if someone looks at your profile) and is abusable (can still post and generate notifications for people with no way for them to view / moderate / report the post) (unless, perhaps there were merge issues and things got dropped, but not sure if that's what happened)

Yea sorry about, I tried to combine several PRs together now, since Ernest added additional PRs fix this his own mistakes/bugs.

According to the RTR2 Kbin PR: 

> This update adds additional user account visibility options and improves the process of deleting user accounts and magazines for the instance administrator. It also introduces private user accounts and magazines similar to options known from Mastodon.
> 
> - ability to delete an account in the user panel
> - additional timestamp indication for the deletion account request
> - ability to withdraw the deletion account request
- cron for automatic deletion of marked accounts (@todo)
> - ability to suspend a user account (the account is not permanently deleted, but threads, posts and comments cease to be visible on the instance)
> - ability to reinstate a user account
> - the new 'deletion' section in the administrator panel, which assists in managing the accounts and magazines marked for deletion
> - responding to a small portion of emails from the contact form

What is not working for you from this list above? Can you go into more details? What errors do you see? Maybe post screenshots? And explain a reproduction scenario.

**Comment by e-five256** [comment]
Sure, I wasn't sure how much you want to dig into it because at this point we're having to debug and fix someone else's code. It's all around suspend account, I haven't tested the mark for deletion stuff yet

I marked my `user1` as suspended.

all of their threads are hidden. comment counts stay the same, but I imagine that is expected even if inaccurate now

image

top level comments are hidden, but replies to comments are **NOT** hidden

image

furthermore, **ALL** of their comments / threads just show up on the user overview page anyways

image

And, as I said, `user`, even suspended, can still post and that still generates notifications for other users, despite the target having no way to stop it / report it or find it, because if they click on it the reply could be hidden

image

**Comment by melroy89** [comment]
Account suspend on the user setting page doesn't make sense, with Mbin we want to improve the delete process.

I will move the account suspend feature to the admin panel instead and fix posts, comments that still shown or the ability to post when the user is set to suspend. As well as fixing the comment counter etc.

**Comment by e-five256** [comment]
suspended accounts can still click the reply button and successfully reply to a comment in a thread or a microblog post

image
image

**Comment by nobodyatroot** [comment]
First "bug" I've encountered with this PR: **both** the delete and purge functions for User content **and** account result in this error being thrown with a fresh Docker test environment using the data test fixtures... memory leak?:

```
Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/mbin/vendor/doctrine/doctrine-bundle/Middleware/BacktraceDebugDataHolder.php on line 45
Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 65536 bytes) in /var/www/mbin/vendor/symfony/error-handler/DebugClassLoader.php on line 296
```

**Comment by melroy89** [comment]
> First "bug" I've encountered with this PR: **both** the delete and purge functions for User content **and** account result in this error being thrown with a fresh Docker test environment using the data test fixtures... memory leak somewhere?:
> 
> ```
> Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/mbin/vendor/doctrine/doctrine-bundle/Middleware/BacktraceDebugDataHolder.php on line 45
> Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 65536 bytes) in /var/www/mbin/vendor/symfony/error-handler/DebugClassLoader.php on line 296
> ```
> 
> EDIT: also include both purge functions under magazine deletion being impacted as well.
> 
> ```
> Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/mbin/src/Service/ActivityPub/Webfinger/WebFingerFactory.php on line 48
> 
> Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 32768 bytes) in /var/www/mbin/vendor/symfony/error-handler/Error/OutOfMemoryError.php on line 1
> 
> Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 1052672 bytes) in /var/www/mbin/vendor/league/mime-type-detection/src/FinfoMimeTypeDetector.php on line 76
> 
> Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 245760 bytes) in /var/www/mbin/vendor/symfony/error-handler/Resources/views/logs.html.php on line 38
> ```

https://stackoverflow.com/questions/415801/allowed-memory-size-of-33554432-bytes-exhausted-tried-to-allocate-43148176-byte xD?

**Comment by melroy89** [comment]
But yea.. maybe we should reduce the memory usage somehow.. I think purge/delete is already doing it in batches.. 😕

File: config/packages/security.yaml [hunk 3]
```diff
@@ -65,215 +65,215 @@ security:
     role_hierarchy:
         ROLE_OAUTH2_WRITE:
             [
-                "ROLE_OAUTH2_ENTRY:CREATE",
```
Comment (e-five256) [inline config/packages/security.yaml:68]: feel like we're going to be going back and forth a lot on this `'` vs `"` thing :laughing:

**Comment by e-five256** [comment]
suspended account content also still appears in search

image

**Comment by melroy89** [comment]
> suspended accounts can still click the reply button and successfully reply to a comment in a thread or a microblog post edit: and also the overview pages of users (basically anywhere inline reply is used)
> 
> image image

yes I know. Not gonna fix it anymore in this massive PR to be honest.

File: config/packages/security.yaml (hunk 3 above)
Comment (melroy89) [reply config/packages/security.yaml:68]: haha yes.. it's final now `.editconfig` says single quote for all YAML files!

**Comment by e-five256** [comment]
I don't think it's a blocker for this PR, as it's the same as how ernest shipped the changes, but it might be good to note it as a known issue. Hopefully something that can be resolved before it gets widely used. I would like to get more input on this PR as I'm not an instance owner and these changes are pretty large, but not sure if who might have time to check it out

**Comment by melroy89** [comment]
@nobodyatroot was I think also testing it. He found a bigger issue I think with 500 error. Not sure yet what the root cause was. Maybe out of memory. Or docker setup related... Anyway I want to find out that bug.

**Comment by melroy89** [comment]
> @nobodyatroot was I think also testing it. He found a bigger issue I think with 500 error. Not sure yet what the root cause was. Maybe out of memory. Or docker setup related... Anyway I want to find out that bug.
> 
> > as it's the same as how ernest shipped the changes,
> 
> Not fully true, he didn't implement any of the warning/errors about "User account is suspended" you are now seeing.

As discussed in Matrix, it was unrelated to this PR.

**Comment by e-five256** [comment]
> Not fully true, he didn't implement any of the warning/errors about "User account is suspended" you are now seeing.

Right, right, of course, I didn't mean that as in you didn't change anything. What I meant was, the **issues** I _am_ seeing come from upstream, they aren't on things you added to it, so that's why I was saying I don't see them as blockers

**Review by nobodyatroot — APPROVED** [review]
Good enough for a sync and integrating Mbin specific things, lots of things can be addressed/fixed in future PRs.

**Review by e-five256 — APPROVED** [review]
we'll want a follow up PR to mbin-ify a few of the things but this at least brings us closer in sync. Not noticing any problems in dev other than what is already known from the comments here

**Pull request merged**